kombu/transport/redis.py

```diff
@@ -1201,7 +1204,10 @@ def _connparams(self, asynchronous=False):
             class Connection(connection_cls):
                 def disconnect(self, *args):
                     super().disconnect(*args)
-                    channel._on_connection_disconnect(self)
+                    # We remove the connection from the poller
```

Comment:
is it possible to add some sort of integration test for this change?

Reply:
@auvipy 
For sure it will be hard to simulate GC in integration tests if we talk about this corner case which we found.
But if we just speak about removing ONLY the channel with _regstered flag the unit test is enough if you ask me.